Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -915,4 +918,31 @@ - (void)editHistoryButtonPushed {
     [self presentViewController:nc animated:YES completion:nil];
 }
 
+#pragma mark - Previewing
+
+- (JSValue*)htmlElementAtLocation:(CGPoint)location {
+    return [[self.webView wmf_javascriptContext][@"getElementFromPoint"] callWithArguments:@[@(location.x), @(location.y)]];
+}
+
+- (MWKTitle*)titleForHTMLElement:(JSValue*)element {
+    if ([[[element valueForProperty:@"tagName"] toString] isEqualToString:@"A"] && [element valueForProperty:@"href"]) {
+        NSURL* url = [NSURL URLWithString:[[element valueForProperty:@"href"] toString]];
+        // Don't peek intra-page fragment links - such as references.
+        if ([url wmf_isIntraPageFragment]) {
+            return nil;
+        }
+        return [[MWKTitle alloc] initWithURL:url];
+    }
```

Comment:
it's tricky, IMO creating a title w/ https://en.wikipedia.org/wiki/Cat is perfectly valid.  even "external links" can go to another wiki, right?  i don't think that `MWKTitle` should implement the "what kind of link this is" business logic, opting instead for maybe the WebVC to implement the following logic for **links on wiki pages**:

- Link is absolute (prefix of `(http(s):)//`)
  - domain contains `wikipedia.org`: create `MWKTitle` and ask delegate handle link tap for title
  - otherwise: open external URL
- Link is a relative path (prefix of `/`): create `MWKTitle` relative to the site where the link was found and ask delegate to handle link tap for title
- Link is a fragment (prefix of `#`): check for conventional prefixes (citation, section, etc.) and open accordingly